### DISCUSSION TITLE
Uploading files to github using api

### DISCUSSION THREAD

**Discussion by shiva-sandupatla** [community/community#24381] [Apps, API and Webhooks]
<p>:wave:<strong>Hi, mates.</strong></p>
<p><strong>Hope you are going well.</strong></p>
<p><strong>I am new to github api, can anyone show an example that how i can upload files to my repository using javascript</strong></p>
<p><strong>Hoping with a positive response</strong> :innocent:</p>
<blockquote>
<p><strong>Regards, $#!v@</strong></p>
</blockquote>

**Comment by nethgato** [comment, marked as answer]
<p>Hi, hi, <a class="mention" href="/shivani-designer">@shivani-designer</a>!</p>
<p>Looks like there’s a very similar conversation over in this thread:</p>
<aside class="quote" data-post="1" data-topic="14031">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-to-upload-a-file-using-github-api/14031">How to upload a file using GitHub API</a> <a class="badge-wrapper  bullet" href="/c/github-api-development-and-support/37"><span class="badge-category-bg" style="background-color: #92278F;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and support using the GitHub API, building GitHub Apps, and everything else to do with the GitHub Platform.">GitHub API Development and Support</span></a>
  </div>
  <blockquote>
    Maybe it’s just me, but this guide <a href="https://developer.github.com/v3/repos/contents/" rel="nofollow noopener">https://developer.github.com/v3/repos/contents/</a> is incredibly vague. 
Simple examples would be handy, and not so much emphasis on what the return messages. 
The objective is to upload a file, preferably .pdf or .png through a curl command. 
Any help/pointers will be highly appreciated.
  </blockquote>
</aside>

<p>Doesn’t look like there’s an explicit Javascript example, but you can take any working example and use something like <a href="https://www.postman.com/">Postman</a> which will kick out a variety of Js options!</p>
<p>If you’re still needing explicit guidance, please let us know!</p>
